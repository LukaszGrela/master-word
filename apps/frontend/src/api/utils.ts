import { TPartialGameState } from './types';

export const createGameState = (
  attempts: number,
  wordLength: number
): TPartialGameState => {
  const word = Array.from(Array(wordLength)).map(() => '');
  return Array.from(Array(attempts)).map(() => ({
    word: word.concat(),
  }));
};
