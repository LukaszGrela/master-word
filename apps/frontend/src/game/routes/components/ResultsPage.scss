.results {
  .game-language {
    margin-bottom: 2rem;
    .language-info {
      margin: 0 0 0.5rem 0;
    }

    .language-selector {
      position: relative;
      justify-content: center;
      top: initial;
      right: initial;

      padding-bottom: 0.25rem;
    }
  }

  .score-container {
    position: relative;
    margin: 0 auto;
    max-width: 20rem;
  }

  .button-row {
    margin-bottom: 4rem;
  }
}
