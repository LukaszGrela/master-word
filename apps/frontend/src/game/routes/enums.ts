export enum EPaths {
  ROOT = '/',
  GAME = '/game',
  RESULTS = '/results/:result',
}

export const getResultsPath = (result: 'win' | 'lose') => `/results/${result}`;
