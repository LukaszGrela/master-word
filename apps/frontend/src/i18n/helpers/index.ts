/* v8 ignore start */
export * from './loadTranslation';
/* v8 ignore stop */
