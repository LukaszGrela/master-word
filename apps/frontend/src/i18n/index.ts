/* v8 ignore start */
export * from './LanguageContext/hooks';
export { LanguageProvider } from './LanguageContext/index';
export type { ILanguageContext } from './LanguageContext/index';
/* v8 ignore stop */
