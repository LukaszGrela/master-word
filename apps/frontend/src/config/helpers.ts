import { AppStorage, EStorageKeys } from '@repo/utils';
import { IConfig } from './types';
import { ATTEMPTS, LANGUAGE, WORD_LENGTH } from '../game';

/**
 * Assure local storage is using enabled values from current config object
 * @param config Current Configuration
 */
export function updateLocalStorageConfig(config: IConfig) {
  console.log('config', config);
  const storage = AppStorage.getInstance();
  // game language
  const configLang =
    config.enabledLanguages.length > 0 ? config.enabledLanguages : [LANGUAGE];
  if (!storage.has(EStorageKeys.GAME_LANGUAGE)) {
    storage.setItem(EStorageKeys.GAME_LANGUAGE, configLang[0]);
  } else {
    // value exists, confirm it is allowed
    if (
      configLang.indexOf(storage.getItem(EStorageKeys.GAME_LANGUAGE)!) === -1
    ) {
      storage.setItem(EStorageKeys.GAME_LANGUAGE, configLang[0]);
    }
  }

  // game word length
  const configLength =
    config.enabledLength.length > 0 ? config.enabledLength : [WORD_LENGTH];
  if (!storage.has(EStorageKeys.GAME_WORD_LENGTH)) {
    storage.setItem(EStorageKeys.GAME_WORD_LENGTH, `${configLength[0]}`);
  } else {
    // value exists, confirm it is allowed
    if (
      configLength.indexOf(
        Number(storage.getItem(EStorageKeys.GAME_WORD_LENGTH)!),
      ) === -1
    ) {
      storage.setItem(EStorageKeys.GAME_WORD_LENGTH, `${configLength[0]}`);
    }
  }

  // game attempts
  const configAttempts =
    config.enabledAttempts.length > 0 ? config.enabledAttempts : [ATTEMPTS];
  if (!storage.has(EStorageKeys.GAME_ATTEMPTS)) {
    storage.setItem(EStorageKeys.GAME_ATTEMPTS, `${configAttempts[0]}`);
  } else {
    // value exists, confirm it is allowed
    if (
      configAttempts.indexOf(
        Number(storage.getItem(EStorageKeys.GAME_ATTEMPTS)!),
      ) === -1
    ) {
      storage.setItem(EStorageKeys.GAME_ATTEMPTS, `${configAttempts[0]}`);
    }
  }

  // UI
  if (!storage.has(EStorageKeys.UI_LANGUAGE)) {
    storage.setItem(EStorageKeys.UI_LANGUAGE, LANGUAGE);
  }
}
