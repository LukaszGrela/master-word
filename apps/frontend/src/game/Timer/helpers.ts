import { toHMS } from '@repo/utils';

const prefix0 = (n: number): string => (n < 10 ? `0${n}` : `${n}`);

export const displayTime = (ms: number): string => {
  const { h, m, s } = toHMS(ms);

  if (!h && !m) {
    return `00:${prefix0(s >> 0)}`;
  }

  if (!h && m !== undefined) {
    return `${prefix0(m)}:${prefix0(s)}`;
  }

  if (h !== undefined && m !== undefined) {
    return `${prefix0(h)}:${prefix0(m)}:${prefix0(s)}`;
    /* v8 ignore start */
  }

  return '00:00'; // it should never reach here
};
/* v8 ignore stop */
