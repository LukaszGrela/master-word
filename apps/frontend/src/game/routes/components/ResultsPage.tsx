import { Location, useLocation, useNavigate } from 'react-router-dom';
import { TGameSessionRecord } from '../../../api';
import { useLanguage } from '../../../i18n';
import { EPaths, getResultsPath } from '../enums';
import { displayTime } from '@repo/utils';
import { useCallback } from 'react';
import { GameLanguage } from '../../language';
import Stars from '../../Stars';

import './ResultsPage.scss';

export const ResultsPage = () => {
  const navigate = useNavigate();
  const { getUIText: t } = useLanguage();
  const location = useLocation() as Location<TGameSessionRecord>;
  const gameSession = location.state;

  const win = location.pathname === getResultsPath('win');

  const playTimeMs = gameSession.game.finished
    ? Number(gameSession.game.timestamp_finish) -
      Number(gameSession.game.timestamp_start)
    : 0;

  const playTime = playTimeMs > 0 ? displayTime(playTimeMs) : '';

  const handleAction = useCallback(
    (action: 'again' | 'quit') => () => {
      if (action === 'quit') {
        navigate(EPaths.ROOT, { replace: true });
      }
      if (action === 'again') {
        navigate(EPaths.GAME, {
          replace: true,
          state: gameSession.session,
        });
      }
    },
    [gameSession.session, navigate],
  );

  let score = 0; // out of 3 max
  if (win) {
    score = 0.5; // max attempts
    const { attempt } = gameSession.game;
    console.log('attempt', attempt);
    if (attempt < 8) {
      score = 1;
    }
    if (attempt < 6) {
      score = 2;
    }
    if (attempt < 4) {
      score = 3;
    }

    // time extras
    const seconds = playTimeMs / 1000;
    console.log(seconds);
    if (seconds <= 15) {
      score += 3;
    }
    if (seconds > 15 && seconds <= 35) {
      score += 2;
    }
    if (seconds > 34 && seconds <= 60) {
      score += 1;
    }
    if (seconds > 60 && seconds <= 90) {
      score += 0.5;
    }
  }

  return (
    <div className="results">
      <h2>
        {win && t('result-win', { attempt: gameSession.game.attempt })}
        {!win && t('result-lose', { wordToGuess: gameSession.game.word })}
      </h2>

      <Stars score={score} />

      {playTimeMs > 0 && (
        <p className="playtime">{t('result-playtime', { playTime })}</p>
      )}

      <GameLanguage />

      <div className="button-row">
        <button onClick={handleAction('quit')} className={'secondary'}>
          {t('result-quit-button')}
        </button>
        <button onClick={handleAction('again')} className={'primary'}>
          {t('result-again-button')}
        </button>
      </div>
    </div>
  );
};
