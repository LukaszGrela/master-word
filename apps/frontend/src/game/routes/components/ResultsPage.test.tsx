import { Mock, afterEach, beforeEach, describe, expect, it, vi } from 'vitest';
import { defaultLanguageContext } from '../../../i18n/LanguageContext/__tests__/contextRenderer';
import { ResultsPage } from './ResultsPage';
import { AppStorage, EStorageKeys, IStorage, hasOwn } from '@repo/utils';
import { IDictionary } from '@repo/common-types';
import { replaceSubstituteMap } from '../../../i18n/helpers';
import { createMemoryRouter } from 'react-router-dom';
import { EPaths, getResultsPath } from '../enums';
import { renderMockRouter } from '../__tests__/helpers';
import { fireEvent, waitFor } from '@testing-library/react';

vi.mock('@repo/utils', async (importOriginal) => {
  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-explicit-any
  const original = await importOriginal<any>();

  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  const AppStorageMock = vi.fn<any, IStorage>();
  const proto = AppStorageMock.prototype as IStorage;

  proto.getItem = vi.fn();
  proto.setItem = vi.fn();
  proto.removeItem = vi.fn();
  proto.has = vi.fn();
  proto.clear = vi.fn();
  proto.clearStorage = vi.fn();

  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
  const instance = new AppStorageMock();

  // eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/no-unsafe-member-access
  (AppStorageMock as any).getInstance = vi.fn().mockReturnValue(instance);

  // eslint-disable-next-line @typescript-eslint/no-unsafe-return
  return {
    ...original,

    AppStorage: AppStorageMock,
  };
});

const originalFetch = global.fetch;

describe('game', () => {
  describe('routes', () => {
    describe('components', () => {
      describe('ResultsPage', () => {
        const gameWon = {
          _id: '660d501fa79a7dd174b8f527',
          session: '590fe9dd-08f7-4fde-90ba-105c9a2ce00b',
          game: {
            language: 'pl',
            timestamp_start: '1712211954608',
            max_attempts: 8,
            attempt: 3,
            word: 'CACKA',
            word_length: 5,
            guessed: true,
            score: 3.25,
            state: [
              {
                word: ['T', 'E', 'R', 'M', 'Y'],
                validated: ['X', 'X', 'X', 'X', 'X'],
                _id: '660e47f2a79a7dd174b8f534',
              },
              {
                word: ['P', 'A', 'C', 'K', 'A'],
                validated: ['X', 'C', 'C', 'C', 'C'],
                _id: '660e481da79a7dd174b8f543',
              },
              {
                word: ['C', 'A', 'C', 'K', 'A'],
                validated: ['C', 'C', 'C', 'C', 'C'],
                _id: '660e4822a79a7dd174b8f54f',
              },
              {
                word: ['', '', '', '', ''],
              },
              {
                word: ['', '', '', '', ''],
              },
              {
                word: ['', '', '', '', ''],
              },
              {
                word: ['', '', '', '', ''],
              },
              {
                word: ['', '', '', '', ''],
              },
            ],
            finished: true,
            _id: '660d501fa79a7dd174b8f528',
            timestamp_finish: '1712212002613',
          },
          __v: 2,
        };
        const gameLost = {
          _id: '660d501fa79a7dd174b8f527',
          session: '590fe9dd-08f7-4fde-90ba-105c9a2ce00b',
          __v: 9,
          highest: {
            'pl:5': {
              score: 3.25,
              timeMs: 48005,
              attempts: 3,
              language: 'pl',
              length: 5,
              _id: '660e4822a79a7dd174b8f550',
            },
          },
          game: {
            language: 'pl',
            timestamp_start: '1712212870373',
            max_attempts: 8,
            attempt: 8,
            word: 'ŁZAWY',
            word_length: 5,
            guessed: false,
            score: 0,
            state: [
              {
                word: ['B', 'E', 'R', 'T', 'A'],
                validated: ['X', 'X', 'X', 'X', 'M'],
                _id: '660e4b86a79a7dd174b8f566',
              },
              {
                word: ['K', 'A', 'R', 'T', 'A'],
                validated: ['X', 'M', 'X', 'X', 'X'],
                _id: '660e4b89a79a7dd174b8f56f',
              },
              {
                word: ['S', 'A', 'N', 'K', 'I'],
                validated: ['X', 'M', 'X', 'X', 'X'],
                _id: '660e4b8fa79a7dd174b8f581',
              },
              {
                word: ['S', 'T', 'A', 'C', 'H'],
                validated: ['X', 'X', 'C', 'X', 'X'],
                _id: '660e4b9aa79a7dd174b8f597',
              },
              {
                word: ['P', 'T', 'A', 'K', 'I'],
                validated: ['X', 'X', 'C', 'X', 'X'],
                _id: '660e4ba1a79a7dd174b8f5b1',
              },
              {
                word: ['W', 'R', 'A', 'C', 'A'],
                validated: ['M', 'X', 'C', 'X', 'X'],
                _id: '660e4baaa79a7dd174b8f5cf',
              },
              {
                word: ['S', 'W', 'A', 'R', 'Y'],
                validated: ['X', 'M', 'C', 'X', 'C'],
                _id: '660e4bb4a79a7dd174b8f5f1',
              },
              {
                word: ['P', 'R', 'A', 'W', 'Y'],
                validated: ['X', 'X', 'C', 'C', 'C'],
                _id: '660e4bc3a79a7dd174b8f626',
              },
            ],
            finished: true,
            _id: '660e4b82a79a7dd174b8f559',
            timestamp_finish: '1712212931548',
          },
        };
        const translationPl: IDictionary<string> = {
          'general-error-title': 'Ożesz kurczaczki!',
          'general-error-message':
            'Przepraszamy, nieoczekiwany błąd jak piorun z nieba spadł i zepsuł zabawę...',
          'general-error-btn-home': 'Zamknij',
          'game-error-subtitle': 'Problem z sesją gry.',
          'translation-info-sr': 'Wybierz język interfejsu',
          'translation-button-en': 'Angielski',
          'translation-button-pl': 'Polski',
          'result-language-selector-btn-title-pl': 'Słowo polskie',
          'result-language-selector-btn-title-en': 'Słowo angielskie',
          'result-win': 'Wygrałeś w {attempt} próbach',
          'result-lose': 'Przegrałeś, słowo do odgadnięcia to {wordToGuess}',
          'result-playtime': 'Czas rozgrywki: {playTime}',
          'result-again-button': 'Odgadnij następne słowo',
          'result-quit-button': 'Zakończ grę',
          'highscore-label': 'Nowy Rekord',
          'result-save-session-button': 'Kontynuuj później',
        };
        let mockFetch = vi.fn();
        let storage: IStorage = AppStorage.getInstance();
        beforeEach(() => {
          mockFetch = vi.fn();
          storage = AppStorage.getInstance();

          (
            storage.getItem as Mock<[key: string], string | null>
          ).mockImplementation((key) => {
            if (key === 'ui-language') {
              return 'pl';
            }
            if (key === 'word-language') {
              return 'pl';
            }
            return null;
          });

          global.fetch = mockFetch;
        });
        afterEach(() => {
          vi.clearAllMocks();
          global.fetch = originalFetch;
        });
        it('renders win', async () => {
          const router = createMemoryRouter([
            {
              path: EPaths.ROOT,
              index: true,
              element: <p>HOME</p>,
            },
            {
              path: EPaths.RESULTS,
              element: <ResultsPage />,
            },
          ]);
          await router.navigate(getResultsPath('win'), {
            state: gameWon,
          });

          const { container } = renderMockRouter(router, {
            languageContext: {
              ...defaultLanguageContext,
              getUIText: (
                textId: string,
                // eslint-disable-next-line @typescript-eslint/no-explicit-any
                substitutes?: IDictionary<any> | undefined,
              ): string => {
                if (hasOwn(translationPl, textId)) {
                  const copy = translationPl[textId];
                  if (substitutes) {
                    return replaceSubstituteMap(copy, substitutes);
                  }
                  return copy;
                }

                return textId;
              },
            },
          });

          expect(container).toMatchSnapshot();
        }, 10000);
        it('renders lose, no game object', async () => {
          const router = createMemoryRouter([
            {
              path: EPaths.ROOT,
              index: true,
              element: <p>HOME</p>,
            },
            {
              path: EPaths.RESULTS,
              element: <ResultsPage />,
            },
          ]);
          await router.navigate(getResultsPath('win'), {
            state: { session: 'session' },
          });

          const { container } = renderMockRouter(router, {
            languageContext: {
              ...defaultLanguageContext,
              getUIText: (
                textId: string,
                // eslint-disable-next-line @typescript-eslint/no-explicit-any
                substitutes?: IDictionary<any> | undefined,
              ): string => {
                if (hasOwn(translationPl, textId)) {
                  const copy = translationPl[textId];
                  if (substitutes) {
                    return replaceSubstituteMap(copy, substitutes);
                  }
                  return copy;
                }

                return textId;
              },
            },
          });

          expect(container).toMatchSnapshot();
        }, 10000);
        it('Won, continue later', async () => {
          const router = createMemoryRouter([
            {
              path: EPaths.ROOT,
              index: true,
              element: <p>HOME</p>,
            },
            {
              path: EPaths.RESULTS,
              element: <ResultsPage />,
            },
          ]);
          await router.navigate(getResultsPath('win'), {
            state: gameWon,
          });

          const { getByText } = renderMockRouter(router, {
            languageContext: {
              ...defaultLanguageContext,
              getUIText: (
                textId: string,
                // eslint-disable-next-line @typescript-eslint/no-explicit-any
                substitutes?: IDictionary<any> | undefined,
              ): string => {
                if (hasOwn(translationPl, textId)) {
                  const copy = translationPl[textId];
                  if (substitutes) {
                    return replaceSubstituteMap(copy, substitutes);
                  }
                  return copy;
                }

                return textId;
              },
            },
          });

          fireEvent.click(
            getByText(translationPl['result-save-session-button']),
          );

          await waitFor(() => {
            expect(router.state.location.pathname).toEqual('/');
            expect(router.state.location.state).toBeNull();
            // eslint-disable-next-line @typescript-eslint/unbound-method
            expect(storage.setItem).toHaveBeenLastCalledWith(
              EStorageKeys.GAME_SESSION,
              gameWon.session,
            );
          });
        }, 10000);
        it('Won, guess another', async () => {
          const router = createMemoryRouter([
            {
              path: EPaths.ROOT,
              index: true,
              element: <p>HOME</p>,
            },
            {
              path: EPaths.GAME,
              element: <p>GAME</p>,
            },
            {
              path: EPaths.RESULTS,
              element: <ResultsPage />,
            },
          ]);
          await router.navigate(getResultsPath('win'), {
            state: gameWon,
          });

          const { getByText } = renderMockRouter(router, {
            languageContext: {
              ...defaultLanguageContext,
              getUIText: (
                textId: string,
                // eslint-disable-next-line @typescript-eslint/no-explicit-any
                substitutes?: IDictionary<any> | undefined,
              ): string => {
                if (hasOwn(translationPl, textId)) {
                  const copy = translationPl[textId];
                  if (substitutes) {
                    return replaceSubstituteMap(copy, substitutes);
                  }
                  return copy;
                }

                return textId;
              },
            },
          });

          fireEvent.click(getByText(translationPl['result-again-button']));

          await waitFor(() => {
            expect(router.state.location.pathname).toEqual(EPaths.GAME);
            expect(router.state.location.state).toMatchObject({session:gameWon.session});
          });
        }, 10000);
        it('renders lose', async () => {
          const router = createMemoryRouter([
            {
              path: EPaths.ROOT,
              index: true,
              element: <p>HOME</p>,
            },
            {
              path: EPaths.RESULTS,
              element: <ResultsPage />,
            },
          ]);
          await router.navigate(getResultsPath('lose'), {
            state: gameLost,
          });

          const { container } = renderMockRouter(router, {
            languageContext: {
              ...defaultLanguageContext,
              getUIText: (
                textId: string,
                // eslint-disable-next-line @typescript-eslint/no-explicit-any
                substitutes?: IDictionary<any> | undefined,
              ): string => {
                if (hasOwn(translationPl, textId)) {
                  const copy = translationPl[textId];
                  if (substitutes) {
                    return replaceSubstituteMap(copy, substitutes);
                  }
                  return copy;
                }

                return textId;
              },
            },
          });

          expect(container).toMatchSnapshot();
        }, 10000);
        it('Lost, finish game', async () => {
          const router = createMemoryRouter([
            {
              path: EPaths.ROOT,
              index: true,
              element: <p>HOME</p>,
            },
            {
              path: EPaths.RESULTS,
              element: <ResultsPage />,
            },
          ]);
          await router.navigate(getResultsPath('lose'), {
            state: gameLost,
          });

          const { getByText } = renderMockRouter(router, {
            languageContext: {
              ...defaultLanguageContext,
              getUIText: (
                textId: string,
                // eslint-disable-next-line @typescript-eslint/no-explicit-any
                substitutes?: IDictionary<any> | undefined,
              ): string => {
                if (hasOwn(translationPl, textId)) {
                  const copy = translationPl[textId];
                  if (substitutes) {
                    return replaceSubstituteMap(copy, substitutes);
                  }
                  return copy;
                }

                return textId;
              },
            },
          });

          fireEvent.click(getByText(translationPl['result-quit-button']));

          await waitFor(() => {
            expect(router.state.location.pathname).toEqual(EPaths.ROOT);
            expect(router.state.location.state).toBeNull();
            // eslint-disable-next-line @typescript-eslint/unbound-method
            expect(storage.removeItem).toHaveBeenLastCalledWith(
              EStorageKeys.GAME_SESSION,
            );
          });
        }, 10000);
      });
    });
  });
});
