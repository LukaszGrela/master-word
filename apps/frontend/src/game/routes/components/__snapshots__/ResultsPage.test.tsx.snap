// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`game > routes > components > ResultsPage > renders lose 1`] = `
<div>
  <div
    class="results"
  >
    <h2>
      Przegrałeś, słowo do odgadnięcia to ŁZAWY
    </h2>
    <div
      class="score-container"
    >
      <div
        class="stars"
      >
        <svg
          class="grade-icon size-medium zero"
          height="24"
          viewBox="0 -960 960 960"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m354-247 126-76 126 77-33-144 111-96-146-13-58-136-58 135-146 13 111 97-33 143ZM233-80l65-281L80-550l288-25 112-265 112 265 288 25-218 189 65 281-247-149L233-80Zm247-350Z"
          />
        </svg>
        <svg
          class="grade-icon size-large zero"
          height="24"
          viewBox="0 -960 960 960"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m354-247 126-76 126 77-33-144 111-96-146-13-58-136-58 135-146 13 111 97-33 143ZM233-80l65-281L80-550l288-25 112-265 112 265 288 25-218 189 65 281-247-149L233-80Zm247-350Z"
          />
        </svg>
        <svg
          class="grade-icon size-medium zero"
          height="24"
          viewBox="0 -960 960 960"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m354-247 126-76 126 77-33-144 111-96-146-13-58-136-58 135-146 13 111 97-33 143ZM233-80l65-281L80-550l288-25 112-265 112 265 288 25-218 189 65 281-247-149L233-80Zm247-350Z"
          />
        </svg>
      </div>
    </div>
    <p
      class="playtime"
    >
      Czas rozgrywki: 1m 1s
    </p>
    <div
      class="game-language"
    >
      <p
        class="language-info"
      >
        result-language-info
      </p>
      <div
        class="language-selector translation"
      >
        <span
          class="hidden sr"
        >
          Wybierz język interfejsu
        </span>
        <button
          class="translation-btn selected"
          title="Słowo polskie"
        >
          <span
            class="label"
          >
            word-language-selector-btn-label-pl
          </span>
          <span
            class="flag"
          >
            <img
              alt=""
              loading="lazy"
              src="https://flagcdn.com/w20/pl.png"
              srcset="https://flagcdn.com/w40/pl.png 2x"
              width="20"
            />
          </span>
        </button>
        <button
          class="translation-btn"
          title="Słowo angielskie"
        >
          <span
            class="label"
          >
            word-language-selector-btn-label-en
          </span>
          <span
            class="flag"
          >
            <img
              alt=""
              loading="lazy"
              src="https://flagcdn.com/w20/us.png"
              srcset="https://flagcdn.com/w40/us.png 2x"
              width="20"
            />
          </span>
        </button>
      </div>
    </div>
    <div
      class="button-row"
    >
      <button
        class="secondary"
      >
        Kontynuuj później
      </button>
      <button
        class="secondary"
      >
        Zakończ grę
      </button>
      <button
        class="primary"
      >
        Odgadnij następne słowo
      </button>
    </div>
  </div>
</div>
`;

exports[`game > routes > components > ResultsPage > renders lose, no game object 1`] = `
<div>
  <div
    class="results"
  >
    <h2>
      Wygrałeś w 0 próbach
    </h2>
    <div
      class="score-container"
    >
      <div
        class="stars"
      >
        <svg
          class="grade-icon size-medium zero"
          height="24"
          viewBox="0 -960 960 960"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m354-247 126-76 126 77-33-144 111-96-146-13-58-136-58 135-146 13 111 97-33 143ZM233-80l65-281L80-550l288-25 112-265 112 265 288 25-218 189 65 281-247-149L233-80Zm247-350Z"
          />
        </svg>
        <svg
          class="grade-icon size-large zero"
          height="24"
          viewBox="0 -960 960 960"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m354-247 126-76 126 77-33-144 111-96-146-13-58-136-58 135-146 13 111 97-33 143ZM233-80l65-281L80-550l288-25 112-265 112 265 288 25-218 189 65 281-247-149L233-80Zm247-350Z"
          />
        </svg>
        <svg
          class="grade-icon size-medium zero"
          height="24"
          viewBox="0 -960 960 960"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m354-247 126-76 126 77-33-144 111-96-146-13-58-136-58 135-146 13 111 97-33 143ZM233-80l65-281L80-550l288-25 112-265 112 265 288 25-218 189 65 281-247-149L233-80Zm247-350Z"
          />
        </svg>
      </div>
      <span
        class="highscore-label"
      >
        Nowy Rekord
      </span>
    </div>
    <div
      class="game-language"
    >
      <p
        class="language-info"
      >
        result-language-info
      </p>
      <div
        class="language-selector translation"
      >
        <span
          class="hidden sr"
        >
          Wybierz język interfejsu
        </span>
        <button
          class="translation-btn selected"
          title="Słowo polskie"
        >
          <span
            class="label"
          >
            word-language-selector-btn-label-pl
          </span>
          <span
            class="flag"
          >
            <img
              alt=""
              loading="lazy"
              src="https://flagcdn.com/w20/pl.png"
              srcset="https://flagcdn.com/w40/pl.png 2x"
              width="20"
            />
          </span>
        </button>
        <button
          class="translation-btn"
          title="Słowo angielskie"
        >
          <span
            class="label"
          >
            word-language-selector-btn-label-en
          </span>
          <span
            class="flag"
          >
            <img
              alt=""
              loading="lazy"
              src="https://flagcdn.com/w20/us.png"
              srcset="https://flagcdn.com/w40/us.png 2x"
              width="20"
            />
          </span>
        </button>
      </div>
    </div>
    <div
      class="button-row"
    >
      <button
        class="secondary"
      >
        Kontynuuj później
      </button>
      <button
        class="secondary"
      >
        Zakończ grę
      </button>
      <button
        class="primary"
      >
        Odgadnij następne słowo
      </button>
    </div>
  </div>
</div>
`;

exports[`game > routes > components > ResultsPage > renders win 1`] = `
<div>
  <div
    class="results"
  >
    <h2>
      Wygrałeś w 3 próbach
    </h2>
    <div
      class="score-container"
    >
      <div
        class="stars"
      >
        <svg
          class="grade-icon gold size-medium full"
          height="24"
          viewBox="0 -960 960 960"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m233-80 65-281L80-550l288-25 112-265 112 265 288 25-218 189 65 281-247-149L233-80Z"
          />
        </svg>
        <svg
          class="grade-icon gold size-large full"
          height="24"
          viewBox="0 -960 960 960"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m233-80 65-281L80-550l288-25 112-265 112 265 288 25-218 189 65 281-247-149L233-80Z"
          />
        </svg>
        <svg
          class="grade-icon gold size-medium full"
          height="24"
          viewBox="0 -960 960 960"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m233-80 65-281L80-550l288-25 112-265 112 265 288 25-218 189 65 281-247-149L233-80Z"
          />
        </svg>
      </div>
      <span
        class="highscore-label"
      >
        Nowy Rekord
      </span>
    </div>
    <p
      class="playtime"
    >
      Czas rozgrywki: 48.005s
    </p>
    <div
      class="game-language"
    >
      <p
        class="language-info"
      >
        result-language-info
      </p>
      <div
        class="language-selector translation"
      >
        <span
          class="hidden sr"
        >
          Wybierz język interfejsu
        </span>
        <button
          class="translation-btn selected"
          title="Słowo polskie"
        >
          <span
            class="label"
          >
            word-language-selector-btn-label-pl
          </span>
          <span
            class="flag"
          >
            <img
              alt=""
              loading="lazy"
              src="https://flagcdn.com/w20/pl.png"
              srcset="https://flagcdn.com/w40/pl.png 2x"
              width="20"
            />
          </span>
        </button>
        <button
          class="translation-btn"
          title="Słowo angielskie"
        >
          <span
            class="label"
          >
            word-language-selector-btn-label-en
          </span>
          <span
            class="flag"
          >
            <img
              alt=""
              loading="lazy"
              src="https://flagcdn.com/w20/us.png"
              srcset="https://flagcdn.com/w40/us.png 2x"
              width="20"
            />
          </span>
        </button>
      </div>
    </div>
    <div
      class="button-row"
    >
      <button
        class="secondary"
      >
        Kontynuuj później
      </button>
      <button
        class="secondary"
      >
        Zakończ grę
      </button>
      <button
        class="primary"
      >
        Odgadnij następne słowo
      </button>
    </div>
  </div>
</div>
`;
